pkg/cloudprovider/vsphere/cloud.go

```diff
@@ -145,13 +163,24 @@ func (vs *VSphere) HasClusterID() bool {
 // Initializes vSphere from vSphere CloudProvider Configuration
 func buildVSphereFromConfig(cfg *CPIConfig) (*VSphere, error) {
 	nm := newNodeManager(cfg, nil)
-
+	lb, err := loadbalancer.NewLBProvider(&cfg.LBConfig)
```

Comment:
Put the gate check here and set lb to nil if it doesn't allow

Reply:
done